### PR TITLE
Fix schema compare script formatting

### PR DESCRIPTION
**Pull request by kisantia**
Add GO after each statement and remove leading and trailing whitespace. Fixes a couple of the problems in https://github.com/microsoft/azuredatastudio/issues/5419

**Adding GO:**
old:
image
new:
image

**No more blank lines at the beginning of some scripts:**
old:
image
fixed:
image

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs [hunk 1]
```diff
@@ -210,10 +210,26 @@ private string GetConnectionString(ConnectionInfo connInfo, string databaseName)
...
+            }
+            return script;
+        }
+
+        public static string FormatScript(string script)
+        {
+            script = RemoveExcessWhitespace(script);
+            if(!string.IsNullOrWhiteSpace(script) && !script.Equals("null"))
+            {
+                script += "\nGO";
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: >\n [](start = 27, length = 2)

Where is this script generated from? Normally you'd use Environment.NewLine in C#

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using Microsoft.SqlTools.ServiceLayer.SchemaCompare;
+using Xunit;
+
+namespace Microsoft.SqlTools.ServiceLayer.UnitTests.SchemaCompare
+{
+    public class SchemaCompareTests
```
Comment (Charles-Gagnon) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: >SchemaCompareTests [](start = 17, length = 18)

yay tests!

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        [Fact]
+        public void FormatScriptAddsGo()
+        {
+            string script = "EXECUTE sp_addextendedproperty @name = N'MS_Description', @value = N'Primary key for AWBuildVersion records.', @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'AWBuildVersion', @level2type = N'COLUMN', @level2name = N'SystemInformationID';";
+            Assert.DoesNotContain("GO", script);
+            string result = SchemaCompareOperation.FormatScript(script);
+            Assert.Contains("GO", result);
```
Comment (Charles-Gagnon) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: nit: But this is redundant with the "EndsWith"

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        [Fact]
+        public void RemovesExcessWhitespace()
+        {
+            // leading whitespace
+            string script1 = "\r\nEXECUTE sp_addextendedproperty @name = N'MS_Description', @value = N'Primary key for AWBuildVersion records.', @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'AWBuildVersion', @level2type = N'COLUMN', @level2name = N'SystemInformationID';";
+            string result1 = SchemaCompareOperation.RemoveExcessWhitespace(script1);
+            Assert.False(script1.Equals(result1));
+            Assert.False(result1.StartsWith("\r"));
```
Comment (Charles-Gagnon) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: You're sure this passes? The regex is stripping spaces, not newlines

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            Assert.DoesNotContain("GO", result);
+            Assert.Equal(string.Empty, result);
+        }
+
+        [Fact]
+        public void RemovesExcessWhitespace()
+        {
+            // leading whitespace
+            string script1 = "\r\nEXECUTE sp_addextendedproperty @name = N'MS_Description', @value = N'Primary key for AWBuildVersion records.', @level0type = N'SCHEMA', @level0name = N'dbo', @level1type = N'TABLE', @level1name = N'AWBuildVersion', @level2type = N'COLUMN', @level2name = N'SystemInformationID';";
+            string result1 = SchemaCompareOperation.RemoveExcessWhitespace(script1);
```
Comment (Charles-Gagnon) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: I'd also add a test that has actual whitespace, not just newlines at the beginning/end

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs (hunk 1 above)
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: >nGO [](start = 28, length = 3)

Is it possible scripts already have GO added to them?

Comment (kisantia) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: Ok changed it to use Environment.NewLine. The script is generated from DacFx

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs (hunk 3 above)
Comment (kisantia) [reply test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: removed

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs (hunk 1 above)
Comment (kisantia) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: No the DacFx script generation doesn't add the GO

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs (hunk 5 above)
Comment (kisantia) [reply test/Microsoft.SqlTools.ServiceLayer.UnitTests/SchemaCompare/SchemaCompareTests.cs]: added a couple spaces to these tests to make sure those get removed too

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs [hunk 6]
```diff
@@ -210,10 +210,26 @@ private string GetConnectionString(ConnectionInfo connInfo, string databaseName)
             return ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
         }
 
-        private static string RemoveExcessWhitespace(string script)
+        public static string RemoveExcessWhitespace(string script)
         {
-            // replace all multiple spaces with single space
-            return Regex.Replace(script, " {2,}", " ");
+            if(script != null)
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: as a style note, we should add a ` ` between `f` and `(` in `if (` statements to be consistent with this codebase.

**Comment by udeeshagautam** [comment]
Not so much of comment but removing extra blank lines - should we leave one and remove extra  or should we remove all. For example in the above ex should we have lines after 8 or 10
image

**Comment by kisantia** [comment]
@udeeshagautam I think it makes sense to take care of the overall formatting of putting the scripts together in ADS
